sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -160,7 +159,7 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
   }
 
   private def assertNoSuchTable(query: String): Unit = {
-    intercept[NoSuchTableException] {
+    intercept[AnalysisException] {
```

Comment:
`AnalysisException` is a pretty general Exception in Spark SQL. The future refactoring might introduce bugs without trigger any test case failure. 

Could we check the error messages?